packages/protocol-identify/src/identify.ts

```diff
@@ -371,6 +252,180 @@ export class Identify implements Startable, IdentifyInterface {
       stream.abort(err)
     }
   }
+}
+
+export class IdentifyPush implements Startable, IdentifyPushInterface {
+  private readonly identifyPushProtocolStr: string
+  public readonly host: {
+    protocolVersion: string
+    agentVersion: string
+  }
+
+  private started: boolean
+  private readonly timeout: number
+  private readonly peerId: PeerId
+  private readonly peerStore: PeerStore
+  private readonly registrar: Registrar
+  private readonly connectionManager: ConnectionManager
+  private readonly addressManager: AddressManager
+  private readonly maxInboundStreams: number
+  private readonly maxOutboundStreams: number
+  private readonly maxIdentifyMessageSize: number
+  private readonly events: TypedEventTarget<Libp2pEvents>
+  private readonly runOnTransientConnection: boolean
+  private readonly log: Logger
+
+  constructor (components: IdentifyPushComponents, init: IdentifyPushInit = {}) {
+    this.started = false
+    this.peerId = components.peerId
+    this.peerStore = components.peerStore
+    this.registrar = components.registrar
+    this.addressManager = components.addressManager
+    this.connectionManager = components.connectionManager
+    this.events = components.events
+    this.log = components.logger.forComponent('libp2p:identify:push')
+
+    this.identifyPushProtocolStr = `/${init.protocolPrefix ?? defaultValues.protocolPrefix}/${MULTICODEC_IDENTIFY_PUSH_PROTOCOL_NAME}/${MULTICODEC_IDENTIFY_PUSH_PROTOCOL_VERSION}`
+    this.timeout = init.timeout ?? defaultValues.timeout
+    this.maxInboundStreams = init.maxInboundStreams ?? defaultValues.maxInboundStreams
+    this.maxOutboundStreams = init.maxOutboundStreams ?? defaultValues.maxOutboundStreams
+    this.maxIdentifyMessageSize = init.maxIdentifyMessageSize ?? defaultValues.maxIdentifyMessageSize
+    this.runOnTransientConnection = init.runOnTransientConnection ?? defaultValues.runOnTransientConnection
+
+    // Store self host metadata
+    this.host = {
+      protocolVersion: `${init.protocolPrefix ?? defaultValues.protocolPrefix}/${IDENTIFY_PROTOCOL_VERSION}`,
+      agentVersion: getAgentVersion(components.nodeInfo, init.agentVersion)
+    }
```

Comment:
It would be nice to consolidate a lot of this into a base identify class that both extend from so they're more manageable.